java/src/jmri/jmrit/logixng/actions/swing/TableForEachSwing.java

```diff
@@ -242,10 +252,10 @@ public void updateObject(@Nonnull Base object) {
             if (_tabbedRowOrColumnPane.getSelectedComponent() == _panelRowOrColumnName) {
                 action.setRowOrColumnAddressing(NamedBeanAddressing.Direct);
                 if (_selectNamedBeanSwing.getAddressing() == NamedBeanAddressing.Direct) {
-                    if (_rowOrColumnNameComboBox.getSelectedIndex() != -1) {
+                    if (_rowOrColumnNameComboBox.getSelectedIndex() > 0) {
```

Comment:
Can 0 be a valid index here?

Reply:
Yes, 0 means the header row or column. A LogixNG table is like an Excel spread sheet where the first row has all the headers for the columns. And the first column has all the headers for the rows.

`_rowOrColumnNameComboBox` has all the row/column headers (depending on if the user has selected "row" or "column"), but the first item in this combo box is `-- Header --` (but possible translated to another language). But internally I use an empty string to mark this since that makes it independent of translations.

So if selected index is 0, it means that the item `-- Header --` is selected and in that case, setRowOrColumnName() should be called with the empty string.

Reply:
By the way, thanks for asking! I appreciate that!